src/components/Property/RaiseRepair/RaiseRepairFormView.js

```diff
@@ -40,15 +46,39 @@ const RaiseRepairFormView = ({ propertyReference }) => {
     setLoading(true)
 
     try {
-      const { id, statusCode, externallyManagedAppointment } = await postRepair(
-        formData
-      )
+      const {
+        id,
+        statusCode,
+        externallyManagedAppointment,
+        externalAppointmentManagementUrl,
+      } = await postRepair(formData)
       setWorkOrderReference(id)
 
       if (statusCode === STATUS_AUTHORISATION_PENDING_APPROVAL.code) {
         setAuthorisationPendingApproval(true)
       } else if (externallyManagedAppointment) {
+        const cookies = new Cookies()
+
+        let schedulerSessionId = cookies.get(
+          process.env.NEXT_PUBLIC_DRS_SESSION_COOKIE_NAME
+        )
+
+        if (!schedulerSessionId) {
+          ;({ schedulerSessionId } = await getSchedulerSessionId())
```

Comment:
Curious what this syntax does here `;({ ... }`

Reply:
This was a new one for me too. It's funky isn't it :) 

As you can see I'm using `let` to assign an initial value on line 63 and then have a need to reassign to that variable, and wanted to use an object destructure when doing so. Apparently JS can't interpret something like `{ myVar } = obj` as it will treat the curlies as a block. So wrapping in parens is a way around that. Then JS needs to also be told that we're not calling a function from a previous line which explains the `;`.

See the 'Assignment without declaration
' section and the note in particular here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#object_destructuring

Reply:
ooh that's neat - thanks for explaining that to me!

Reply:
Once I added the parens, the linter took care of the `;` for me 🙂